### ISSUE TITLE
Refactor report generation logic into a separate repository

### ISSUE THREAD

**Issue by sjperkins** [dask/distributed#6079]
The following file contains report generation logic that could potentially be utilised by multiple testing and benchmarking efforts

https://github.com/dask/distributed/blob/main/continuous_integration/scripts/test_report.py

It may be beneficial to split report generation logic into a separate repository:

/cc @fjetter @jrbourbeau @ncclementi @crusaderky @jcrist @bryanwweber @ian-r-rose @phobson

**Comment by fjetter** [comment]
I'm not sure if this is worth setting up another repository. Even if it is in an extra repository, how would you reuse the code? git submodules? Would you actually package it to be installable?

**Comment by sjperkins** [comment]
> I'm not sure if this is worth setting up another repository. Even if it is in an extra repository, how would you reuse the code? git submodules? Would you actually package it to be installable?

Perhaps it would be useful if it lived in the dask/dask repository? Then it could be available to all benchmarks that use dask.

**Comment by fjetter** [comment]
I don't have a strong opinion on this. My concern is that I don't want anybody worry about "public API" when changing CI tooling and scripts like these. It's just a few hundred lines of code. I don't think vendoring/copying is a big deal.

**Comment by sjperkins** [comment]
> It's just a few hundred lines of code. I don't think vendoring/copying is a big deal.

Yes, maybe its not worth identifying common elements at this point.
